### ISSUE TITLE
PositionLength not returned when Shingle filter is used via Analyze

### ISSUE THREAD

**Issue by titusf** [elastic/elasticsearch-net#5498]
**NEST/Elasticsearch.Net version**: 7.11.1

**Elasticsearch version**: 7.11.2

**Description of the problem including expected versus actual behavior**: When requesting an AnalyzeResponse using a shingle filter, positionLength on each token is null. ElasticSearch returns correct positionLength but NEST does not.

**Steps to reproduce**:
 1. Run the following in NEST: 
```
AnalyzeResponse analyzeResponse = client.Indices.Analyze(a => a
                .Tokenizer("standard")
                .Filter(f => f.Shingle(sh => sh.OutputUnigrams(true)))
                .Text("Example string for debugging")
            );
```
 2. Check each AnalyzeToken on the response. PositionLength will always be null
 3. Check the debug output. positionLength will be present

**Provide `ConnectionSettings` (if relevant)**:

**Provide `DebugInformation` (if relevant)**:
```
# Response:
{"tokens":[{"token":"Example","start_offset":0,"end_offset":7,"type":"<ALPHANUM>","position":0},{"token":"Example string","start_offset":0,"end_offset":14,"type":"shingle","position":0,"positionLength":2},{"token":"string","start_offset":8,"end_offset":14,"type":"<ALPHANUM>","position":1},{"token":"string for","start_offset":8,"end_offset":18,"type":"shingle","position":1,"positionLength":2},{"token":"for","start_offset":15,"end_offset":18,"type":"<ALPHANUM>","position":2},{"token":"for debugging","start_offset":15,"end_offset":28,"type":"shingle","position":2,"positionLength":2},{"token":"debugging","start_offset":19,"end_offset":28,"type":"<ALPHANUM>","position":3}]}
```

**Comment by stevejgordon** [comment]
@titusf Thanks for reporting this. I can see why this is happening and how to resolve it. I'll pick this up with a view to getting it into the next release.

**Comment by stevejgordon** [comment]
@titusf This has just been included in the 7.12.1 release.
